web-api/src/storage/elastic.rs

```diff
@@ -231,32 +224,54 @@ struct Hit<T> {
     score: f32,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Debug, Deserialize)]
 struct Hits<T> {
     hits: Vec<Hit<T>>,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Debug, Deserialize)]
 struct SearchResponse<T> {
     hits: Hits<T>,
 }
 
-/// Represents a document with calculated embeddings that is stored in Elastic Search.
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct Document {
-    pub snippet: String,
-    pub properties: DocumentProperties,
-    pub embedding: Embedding,
-    pub tags: Vec<String>,
+#[derive(Debug, Deserialize)]
+struct InteractedDocument {
+    embedding: Embedding,
+    #[serde(default)]
+    tags: Vec<String>,
+}
+
+impl From<SearchResponse<InteractedDocument>> for Vec<models::InteractedDocument> {
+    fn from(response: SearchResponse<InteractedDocument>) -> Self {
+        response
+            .hits
+            .hits
+            .into_iter()
+            .map(|hit| models::InteractedDocument {
+                id: hit.id,
+                embedding: hit.source.embedding,
+                tags: hit.source.tags,
+            })
+            .collect()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct PersonalizedDocument {
+    #[serde(default)]
+    properties: DocumentProperties,
+    embedding: Embedding,
+    #[serde(default)]
+    tags: Vec<String>,
```

Comment:
Couldn't we just use `[models](models::PersonalizedDocument)` like before, this would keep things a bit more simple, less indirection and confusions about what  does serialize in which way.

Reply:
Same for IngestedDocument.

Reply:
i'd say no for a few reasons:
- it is the specific purpose of this task to minimize traffic in elastic queries, so all unnecessary fields are removed in `elastic::PersonalizedDocument` compared to the `models::PersonalizedDocument` which has additional fields
- the `models::PersonalizedDocument` is the general kind of a personalized document used in storage contexts, whereas the `elastic::PersonalizedDocument` is the specific kind for the elastic storage, depending on the concrete storage impl we might also require further personalized document kinds in the future

Reply:
I still believe it would be better to only have `models::PersonalizedDocument` and `models::InteractedDocument`  at least as long as the storage representation does (mostly) match the runtime representation, which I think we always should keep matching.

I also still don't think having more then one storage engine was a good idea, or that we ever should add another one.

Reply:
> I still believe it would be better to only have models::PersonalizedDocument and models::InteractedDocument at least as long as the storage representation does (mostly) match the runtime representation, which I think we always should keep matching.

it would be nicer, but it's not possible. even if we only had elastic as storage and no storage abstractions, we would still need different types, for example the id field is retrieved in a different structure layout than we require according to the openapi spec. also it wouldn't be possible to save traffic, which is the whole purpose of this task.